apps/src/applab/firebaseStorage.js

```diff
@@ -249,36 +254,51 @@ FirebaseStorage.deleteRecord = function (tableName, record, onComplete, onError)
  * @param {string} tableName Table to listen to.
  * @param {function (Object, RecordListener.EventType)} onRecord Callback to call when
  * a change occurs with the record object (described above) and event type.
+ * @param {function (string)} onWarning Callback to call with an warning to show to the user.
  * @param {function (string, number)} onError Callback to call with an error to show to the user and
  *   http status code.
  */
-FirebaseStorage.onRecordEvent = function (tableName, onRecord, onError) {
+FirebaseStorage.onRecordEvent = function (tableName, onRecord, onWarning, onError, includeAll) {
   if (typeof onError !== 'function') {
     throw new Error('onError is a required parameter to FirebaseStorage.onRecordEvent');
   }
   if (!tableName) {
     onError('Error listening for record events: missing required parameter "tableName"', 400);
     return;
   }
+  if (listenedTables.includes(tableName)) {
+    onWarning(`onRecordEvent was already called for table "${tableName}". To avoid 
+unexpected behavior in your program, you should only call onRecordEvent once 
+per table and use if/else statements to handle the different event types.`);
```

Comment:
unless it's important that we have line breaks in exactly these places in our output string, i think i'd prefer to have this be something like the following (so that we don't end up with this weird indenting in our code)
```
onWarning(`onRecordEvent was already called for table "${tableName}". To avoid` + 
  'unexpected behavior in your program, you should only call onRecordEvent once' +
  'per table and use if/else statements to handle the different event types.');